ci/check_symbols.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+set -eEuo pipefail
+
+echo "checking for symbol visibility issues"
+
+WHEEL_FILE=${1}
+
+raise-symbols-found-error() {
+    local pattern="${1}"
+
+err_msg="ERROR: Found some exported symbols matching the pattern '${pattern}'.
+
+These should be marked with 'hidden' visibility.
+See https://cmake.org/cmake/help/latest/prop_tgt/LANG_VISIBILITY_PRESET.html and https://gcc.gnu.org/wiki/Visibility for details.
+"
+
+    echo ""
+    echo "${err_msg}"
+    exit 1
+}
+
+WHEEL_EXPORT_DIR="$(mktemp -d)"
+
+unzip \
+    -d "${WHEEL_EXPORT_DIR}" \
+    "${WHEEL_FILE}"
+
+dso_files=$(
+    find \
+        "${WHEEL_EXPORT_DIR}" \
+        -type f \
+        \( -name '*.so' -o -name '*.so.*' \)
```

Comment:
Does this work?
```suggestion
        \( -name '*.so*' \)
```

Reply:
I'm sure it would, but it'd also catch `*.sop`, `*.sox`, `*.sock`, etc. Not that any of those are real extensions I know about, but I think the slight extra complexity of the `-o` is worth it here in exchange for a bit more certainty about what's being matched.

I don't feel strongly, so if you find that unconvincing I'll happily change this.

Reply:
I agree with @jameslamb, we should keep this explicit and not accidentally pull in other extensions. Let's keep it the way it is.

Reply:
Makes sense, that's fine with me.